packages/core/src/new-api/types/execution-state.ts

```diff
@@ -120,32 +120,63 @@ export interface DeploymentExecutionState
   > {
   storedArtifactPath: string; // As stored in the deployment directory.
   storedBuildInfoPath?: string; // As stored in the deployment directory. Optional as it's not always present
+  contractName: string;
   constructorArgs: ArgumentType[];
   libraries: Record<string, string>; // TODO: Do we need to store their future ids for the reconciliation process?
   value: bigint;
+  from: string | undefined;
   contractAddress?: string; // The result
+}
 
-  // from: string; -- Will we support this in the beta? I'll ignore it for now, but we should decide it asap
+export interface CallExecutionState
+  extends BaseExecutionState<FutureType.NAMED_CONTRACT_CALL> {
+  contractAddress: string;
+  functionName: string;
+  args: ArgumentType[];
+  value: bigint;
+  from: string | undefined;
 }
 
-type CallExecutionState = BaseExecutionState<FutureType.NAMED_CONTRACT_CALL>;
+export interface StaticCallExecutionState
+  extends BaseExecutionState<FutureType.NAMED_STATIC_CALL> {
+  contractAddress: string;
+  functionName: string;
+  args: ArgumentType[];
+  result?: ArgumentType;
+  from: string | undefined;
+}
 
-type StaticCallExecutionState =
-  BaseExecutionState<FutureType.NAMED_STATIC_CALL>;
+export interface ContractAtExecutionState
+  extends BaseExecutionState<
+    FutureType.NAMED_CONTRACT_AT | FutureType.ARTIFACT_CONTRACT_AT
+  > {
+  contractName: string;
+  address: string;
+}
 
-type ContractAtExecutionState = BaseExecutionState<
-  FutureType.NAMED_CONTRACT_AT | FutureType.ARTIFACT_CONTRACT_AT
->;
+export interface ReadEventArgumentExecutionState
+  extends BaseExecutionState<FutureType.READ_EVENT_ARGUMENT> {
+  eventName: string;
```

Comment:
This lacks the information about which transaction the event should be read from.

Reply:
I am not following. Do you mean the execution state should also record the transactionId for the transaction that emits the event?

Reply:
Yes, or maybe just the id of the future that lead to that transaction, but right now there's no way to extract the event's arg because you can't access the event.